File: src/main/java/com/ninja/ieumnori/ieumnoriserver/banner/domain/Banner.java
package com.ninja.ieumnori.ieumnoriserver.banner.domain;

import com.ninja.ieumnori.ieumnoriserver.common.domain.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Entity
@Table(name = "banner")
public class Banner extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Builder.Default
    @Column(name = "is_active", nullable = false)
    private boolean isActive = true;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "image_url", nullable = false)
    private String imageUrl;

    @Column(name = "redirect_url", nullable = false)
    private String redirectUrl;

    public void update(String title, String imageUrl, String redirectUrl) {
        this.title = title;
        this.imageUrl = imageUrl;
        this.redirectUrl = redirectUrl;
    }

    public void deactivate() {
        this.isActive = false;
    }
}

File: src/main/java/com/ninja/ieumnori/ieumnoriserver/banner/domain/Banner.kt
package com.ninja.ieumnori.ieumnoriserver.banner.domain

import com.ninja.ieumnori.ieumnoriserver.common.domain.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "banner")
class Banner (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @Column(name="title", nullable = false)
    var title: String,
    @Column(name="link", nullable = true)
    var link: String?,
    @Column(name="main_image", nullable = false)
    var mainImage: String,
    @Column(name = "is_exposed", nullable = false)
    var isExposed: Boolean,
): BaseEntity() {
    constructor(): this(0, "", "", "", false)
}